### PR TITLE
Classes probably shouldn't be in org.apache.spark.* and org.apache.hadoop.* namespaces

### PR DESCRIPTION
**Pull request by HyukjinKwon**
https://github.com/databricks/spark-xml/issues/7

1. I just renamed the packages from `org.apache.*` to `com.databricks.*`.

2. Polished imports and reordered them.

3. Updated some accessors to stay independent.

4. Just removed `coverage` which currently does not anything at `.travis.yml`

5. At `InferSchema`, removed `DecimalType` check (it is okay because this package does not infer `DecimalType`), so that it can be independent as`IntergrelType` is not used anymore which has an accessor for only `sql` package in Spark.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@JoshRosen Would you look though this please?

File: src/main/resources/META-INF/services/org.apache.spark.sql.sources.DataSourceRegister [hunk 1]
```diff
@@ -1 +1 @@
-org.apache.spark.sql.xml.DefaultSource15
\ No newline at end of file
+com.databricks.spark.xml.DefaultSource15
```
Comment (HyukjinKwon) [inline src/main/resources/META-INF/services/org.apache.spark.sql.sources.DataSourceRegister]: I already made some branches to open PRs after this PR.
In my local branch (for adding a short name test), I added a newline here.

File: src/main/java/com/databricks/hadoop/mapreduce/lib/input/XmlInputFormat.java [hunk 2]
```diff
@@ -29,10 +30,11 @@
 import org.apache.hadoop.mapreduce.InputSplit;
 import org.apache.hadoop.mapreduce.RecordReader;
 import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
```
Comment (HyukjinKwon) [inline src/main/java/com/databricks/hadoop/mapreduce/lib/input/XmlInputFormat.java]: In alphabetic ordering, this `org.apache.hadoop.mapreduce.lib` should go above but I will update them when downgrading the API version in another PR for Hadoop version compatibility, which I will open right after closing this.

File: .travis.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ matrix:
       scala: 2.11.7
       env: TEST_SPARK_VERSION="1.5.0"
 script:
-  - sbt -Dspark.testVersion=$TEST_SPARK_VERSION ++$TRAVIS_SCALA_VERSION coverage test
```
Comment (JoshRosen) [inline .travis.yml]: Why remove this?

Comment (JoshRosen) [inline .travis.yml]: Even if Codecov isn't set up, this should still end up printing coverage metrics to the console, so there's no harm in leaving it in.

Comment (HyukjinKwon) [inline .travis.yml]: I just wanted to set up this later and properly for this issue https://github.com/databricks/spark-xml/issues/1.

Comment (HyukjinKwon) [inline .travis.yml]: Can I get that back at another PR for that? I will open them right after closing this.

Comment (JoshRosen) [inline .travis.yml]: Sure. Let me go ahead and merge this now. FYI, we should use the `./dev/merge_pr.py` script to merge pull requests and _not_ the GitHub merge button (similar to what we do in our other repos and in Spark itself).

**Comment by JoshRosen** [comment]
Merged. Thanks!

File: .travis.yml (hunk 3 above)
Comment (HyukjinKwon) [inline .travis.yml]: Thanks, I also definitely will.

**Pull request closed without merging**